packages/core/src/components/organisms/DateInput/DateInput.tsx

```diff
@@ -33,24 +34,138 @@ import {
 import {DatePicker} from '../../molecules';
 import DateInputUtils from './date-input.helper';
 
+const ACTION_ICON_SIZE = 30;
+
+interface DateInputSelectionProps {
+  mode: 'date' | 'datetime' | 'time';
+  nullable: boolean;
+  popup: boolean;
+  currentDate: Date;
+  selectedDate: Date;
+  setSelectedDate: (date: Date) => void;
+  onDateChange: (date: Date) => void;
+  setPickerIsOpen: (pickerIsOpen: boolean) => void;
+}
+
+const DateInputSelection = ({
+  mode,
+  nullable,
+  popup,
+  currentDate,
+  selectedDate,
+  setSelectedDate,
+  onDateChange,
```

Comment:
add default value to avoid npe

Reply:
Default values are set on DateInput component, I think there is no need to repeat them

Reply:
onDateChange is mandatory so it's not necessary